### PR TITLE
refactor: unify servers

### PR DESCRIPTION
**Pull request by patte**
- [x] move republik and publikator server into packages
- [x] create an unified `server/graphql` that provides all existing APIs
- [x] move all envs into the root `.env` (`servers/assets` keeps it's own `.env`)
- [x] adapt tests
- [x] cleanup `SERVER` env var.

Migrate prod:
- [x] merge publikator-api ENVs into republik-api
- [ ] deploy
- [ ] remove `SERVER` env var from republik-api
- [ ] set `API_URL=https://api.republik.ch/graphql` and `API_WS_URL=wss://api.republik.ch/graphql` in republik-publikator app
- [ ] revert commit c595288ac99ae5231a703adf7b1392e34f733e15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,25 +107,24 @@ yarn run dev
 ```
 
 This kicks on [foreman](https://github.com/strongloop/node-foreman) which then launches all the servers locally.
-All servers greets you with `Cannot GET /` on the root route. The API servers have a graphical API explorer available at `/graphiql`:
+All servers greets you with `Cannot GET /` on the root route. The API server has a graphical API explorer available at `/graphiql`:
 
-- [Republik API on port 5000](http://localhost:5000/graphiql)
-- [Publikator API on port 5010](http://localhost:5010/graphiql)
+- [Republik & Publikator API on port 5000](http://localhost:5000/graphiql)
```
Comment (caillou) [inline README.md:112]:
```suggestion
- [Republik & Publikator API on port 5000](http://localhost:5000/graphiql)
+ [GraphQL API on port 5000](http://localhost:5000/graphiql)
```

File: README.md [hunk 2]
```diff
@@ -109,25 +107,24 @@ yarn run dev
...
 - [Assets server on port 5020](http://localhost:5020/)
 
 ### Next steps
 
 #### more about ENVs
 
-In development environment variables are loaded from `./.env`, `../../.env` (in this order, relative to node's cwd). This is done with the help of [backend-modules-env](packages/env) and enables you to define a `.env` file at the root of this repo for common variables and also store server specific envs at `servers/*/.env`. `./.env` takes precedence over `../../.env` because existing env variables are never overwritten.
+In development environment variables are loaded from `./.env`r.
```
Comment (caillou) [inline README.md:119]:
```suggestion
In development environment variables are loaded from `./.env`.
```

**Comment by caillou** [comment]
There are still some references to `servers/republik` and `servers/publikator` in the project. While most are commented out, the following need fixing:

```
package.json
58:    "populate:membershipStats:evolution": "servers/republik/script/MembershipStats/evolution/populate.js",
59:    "populate:membershipStats:lastSeen": "servers/republik/script/MembershipStats/lastSeen/populate.js",
68:    "db:seed": "servers/republik/seeds/script/seed.js",
69:    "sendNotification": "cd servers/republik && node ../../packages/notifications/script/sendNotification.js",

packages/voting/scripts/mainstreamscore/seed.js
12:const { loaderBuilders } = require('../../../../servers/republik/server')
```

<details><summary>Here is the full list:</summary>
<p>

```bash
❯ ag "servers/(publikator|republik)"
README.md
131:    - `MAILCHIMP_*` in `servers/republik/.env` (less important)

package.json
58:    "populate:membershipStats:evolution": "servers/republik/script/MembershipStats/evolution/populate.js",
59:    "populate:membershipStats:lastSeen": "servers/republik/script/MembershipStats/lastSeen/populate.js",
68:    "db:seed": "servers/republik/seeds/script/seed.js",
69:    "sendNotification": "cd servers/republik && node ../../packages/notifications/script/sendNotification.js",

packages/republik/migrations/sqls/20180402175144-fix-membership-trigger-down.sql
8:-- to restore the original trigger, see: servers/republik/migrations/sqls/20171127165339-recurring-membership-periods-up.sql:64

packages/republik/script/prolong/mcToken.js
6: * cd servers/republik

packages/republik/script/prolong/printEmails.js
8: * cd servers/republik

packages/republik/script/prolong/mcOwnerNotice.js
7: * Usage: (run from servers/republik)

packages/republik/script/prolong/mcRegular.js
7: * Usage: (run from servers/republik)

packages/republik/script/prolong/mcBonus.js
8: * Usage: (run from servers/republik)

packages/republik/modules/crowdfundings/express/paymentWebhooks.js
79:    // accepted status see: servers/republik/modules/crowdfundings/lib/payments/postfinance/payPledge.js

packages/republik/modules/crowdfundings/lib/payments/postfinance/payPledge.js
35:  // see: servers/republik/modules/crowdfundings/express/paymentWebhooks.js

packages/publikator/script/removeDialogBoxes.js
7: * Usage: (run from servers/publikator)

packages/publikator/script/lib/transformPublications.js
11: * Usage example: (run from servers/publikator)

packages/publikator/script/addDurations.js
6: * Usage: (run from servers/publikator)

packages/voting/scripts/mainstreamscore/seed.js
12:const { loaderBuilders } = require('../../../../servers/republik/server')

packages/documents/lib/meta.js
61: * Publish uses servers/publikator/lib/Document.prepareMetaForPublish instead
```

</p>
</details>

File: .docker-config/traefik/traefik-provider.toml [hunk 3]
```diff
@@ -37,7 +37,7 @@
     [[http.services.api.loadBalancer.servers]]
       url = "http://localhost:5000"
     [[http.services.publikator-api.loadBalancer.servers]]
-      url = "http://localhost:5010"
```
Comment (caillou) [inline .docker-config/traefik/traefik-provider.toml:40]: Wouldn't it make sense to just merge the following services: `api` and `publikator-api`

File: Procfile-dev [hunk 4]
```diff
@@ -1,5 +1,3 @@
-republik: cd servers/republik && PORT=5000 yarn run dev
-republik-scheduler: cd servers/republik && PORT=5000 yarn run devScheduler
-publikator: cd servers/publikator && PORT=5010 yarn run dev
-publikator-scheduler: cd servers/publikator && PORT=5010 yarn run devScheduler
+graphql: cd servers/graphql && PORT=5000 yarn run dev
+scheduler: cd servers/graphql && PORT=5000 yarn run devScheduler
```
Comment (caillou) [inline Procfile-dev:2]: Taken that graphql and scheduler both run on port 5000, aren't these going to conflict?

Comment (patte) [reply Procfile-dev]: The scheduler actually doesn't listen on any port, I removed the var.

File: .docker-config/traefik/traefik-provider.toml (hunk 3 above)
Comment (patte) [reply .docker-config/traefik/traefik-provider.toml:40]: Traefik is only used for development and also there only by me atm. As it does no harm exposing the backend on two ports I'm going to leave it for now.

**Comment by patte** [comment]
> There are still some references to `servers/republik` and `servers/publikator` in the project. While most are commented out, the following need fixing:

Thank you very much for catching those, I cleaned it all up.

File: README.md [hunk 5]
```diff
@@ -80,11 +80,9 @@ Note:
 
 ### 2. ENVs
 
-Copy the `.env.example` files to `.env` (in root and server/\*/). The default values should be enough to get started.
+Copy the `.env.example` files to `.env` (in root and server/assets/). The default values should be enough to get started.
```
Comment (caillou) [inline README.md:83]:
```suggestion
Copy the `.env.example` files to `.env` (in root and servers/assets/). The default values should be enough to get started.
```

File: package.json [hunk 6]
```diff
@@ -65,8 +65,8 @@
     "db:migrate:create": "packages/migrations/script/create.js",
     "db:migrate:up": "db-migrate up --migrations-dir packages/migrations/migrations",
     "db:migrate:down": "db-migrate down --migrations-dir packages/migrations/migrations --verbose",
-    "db:seed": "servers/republik/seeds/script/seed.js",
-    "sendNotification": "cd servers/republik && node ../../packages/notifications/script/sendNotification.js",
+    "db:seed": "packages/republik/seeds/script/seed.js",
+    "sendNotification": "cd packages/republik && node ../../packages/push-notifications/script/sendNotification.js",
```
Comment (caillou) [inline package.json:69]:
```suggestion
    "sendNotification": "cd packages/republik && node ../push-notifications/script/sendNotification.js",
```

File: packages/mail/integration.jest.js [hunk 7]
```diff
@@ -1,11 +1,5 @@
 const path = require('path')
-const relativeServerPath = '../../servers/republik/'
-require('@orbiting/backend-modules-env').config(
-  path.join(__dirname, relativeServerPath, '.test.env')
-)
-require('@orbiting/backend-modules-env').config(
-  path.join(__dirname, relativeServerPath, '.env')
-)
+require('@orbiting/backend-modules-env').config()
```
Comment (caillou) [inline packages/mail/integration.jest.js:2]: Isn't the `.test.env` needed here anymore?

```suggestion
require('@orbiting/backend-modules-env').config('.test.env')
```

File: packages/test/instance/index.js [hunk 8]
```diff
@@ -11,25 +11,16 @@ const { getId } = require('./pool')
...
 
   // load env of server
-  require('@orbiting/backend-modules-env').config(
-    path.join(__dirname, relativeServerPath, '.test.env')
-  )
-  require('@orbiting/backend-modules-env').config(
-    path.join(__dirname, relativeServerPath, '.env')
-  )
+  require('@orbiting/backend-modules-env').config('.test.env')
+  require('@orbiting/backend-modules-env').config('.env')
```
Comment (caillou) [inline packages/test/instance/index.js:23]: Isn't the `.env` loaded by default in `config()`?

```suggestion
  require('@orbiting/backend-modules-env').config('.test.env')
```

File: packages/publikator/script/delete_test_repos.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 process.env.OVERWRITE_ENV = '.test.env'
```
Comment (caillou) [inline packages/publikator/script/delete_test_repos.js:1]: This isn't needed anymore, is it?

```suggestion
```

File: packages/publikator/script/lib/transformPublications.js [hunk 10]
```diff
@@ -8,10 +8,10 @@
  * For each doc transform returns true, the changes are saved to git and
  * the article get's republished.
  *
- * Usage example: (run from servers/publikator)
- * node script/removeDialogBoxes.js [-n [num]] [--dry]
- * -n: stop after n fixes
- * -- dry: don't commit/publish
+ * Usage example:
+ * packages/republik/script/removeDialogBoxes.js [-n [num]] [--dry]
```
Comment (caillou) [inline packages/publikator/script/lib/transformPublications.js:12]:
```suggestion
 * node packages/republik/script/removeDialogBoxes.js [-n [num]] [--dry]
```

File: packages/publikator/README.md [hunk 11]
```diff
@@ -1,47 +1,9 @@
-Publikator Backend
+Publikator
 ------------------
 
-Publikator is a cms prototype: edit files on github with [slate](https://github.com/ianstormtaylor/slate).
+Here you find the code resposible for the CMS tasks.
```
Comment (caillou) [inline packages/publikator/README.md:4]:
```suggestion
Here you find the code responsible for the CMS tasks.
```

File: packages/publikator/package.json [hunk 12]
```diff
@@ -1,11 +1,20 @@
...
+  "name": "@orbiting/backend-modules-publikator",
   "version": "0.0.1",
-  "engines": {
-    "node": "14.x"
+  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/orbiting/backends.git"
   },
-  "private": true,
```
Comment (caillou) [inline packages/publikator/package.json:7]: Why isn't this a `private` repo? Do we plan to publish it to `npm`?

https://docs.npmjs.com/files/package.json#private

File: packages/publikator/script/addDurations.js [hunk 13]
```diff
@@ -3,10 +3,10 @@
  * This script adds duration to the embeds and audioSource of
  * all published articles.
  *
- * Usage: (run from servers/publikator)
- * node script/addDurations.js [-n [num]] [--dry]
- * -n: stop after n fixes
- * -- dry: don't commit/publish
+ * Usage:
+ * packages/publikator/script/addDurations.js [-n [num]] [--dry]
```
Comment (caillou) [inline packages/publikator/script/addDurations.js:7]:
```suggestion
 * node packages/publikator/script/addDurations.js [-n [num]] [--dry]
```

File: packages/publikator/script/removeDialogBoxes.js [hunk 14]
```diff
@@ -4,10 +4,10 @@
  * "Dialog mit der Redaktion" and "Feedback ans Feulleton"
  * from existing published articles.
  *
- * Usage: (run from servers/publikator)
- * node script/removeDialogBoxes.js [-n [num]] [--dry]
- * -n: stop after n fixes
- * -- dry: don't commit/publish
+ * Usage:
+ * packages/publikator/script/removeDialogBoxes.js [-n [num]] [--dry]
```
Comment (caillou) [inline packages/publikator/script/removeDialogBoxes.js:8]:
```suggestion
 * node packages/publikator/script/removeDialogBoxes.js [-n [num]] [--dry]
```

File: packages/republik/script/prolong/mcBonus.js [hunk 15]
```diff
@@ -5,8 +5,8 @@
  * - the BONUS tag for who is eligable
  *
  *
- * Usage: (run from servers/republik)
- * node script/prolong/mcBonus.js
+ * Usage:
+ * packages/republik/script/prolong/mcBonus.js
```
Comment (caillou) [inline packages/republik/script/prolong/mcBonus.js:9]:
```suggestion
 * node packages/republik/script/prolong/mcBonus.js
```

File: packages/republik/script/prolong/mcOwnerNotice.js [hunk 16]
```diff
@@ -4,8 +4,8 @@
  * - if they need to prolong before 2019-01-16
  * - if they to not already have an owner notice key in the maillog
  *
- * Usage: (run from servers/republik)
- * node script/prolong/mcOwnerNotice.js [--dry]
+ * Usage:
+ * packages/republik/script/prolong/mcOwnerNotice.js [--dry]
```
Comment (caillou) [inline packages/republik/script/prolong/mcOwnerNotice.js:8]:
```suggestion
 * node packages/republik/script/prolong/mcOwnerNotice.js [--dry]
```

File: packages/republik/script/prolong/mcRegular.js [hunk 17]
```diff
@@ -4,8 +4,8 @@
  * - a CP_ATOKEN2 if they need to prolong before 2019-01-16
  *
  *
- * Usage: (run from servers/republik)
- * node script/prolong/mcRegular.js [--dry]
+ * Usage:
+ * packages/republik/script/prolong/mcRegular.js [--dry]
```
Comment (caillou) [inline packages/republik/script/prolong/mcRegular.js:8]:
```suggestion
 * node packages/republik/script/prolong/mcRegular.js [--dry]
```

File: packages/republik/script/prolong/mcToken.js [hunk 18]
```diff
@@ -3,8 +3,7 @@
  * This script adds a CP_ATOKEN to mailchimp members to all that can prolong
  *
  * Usage:
- * cd servers/republik
- * script/prolong/mcToken.js --dry
+ * packages/republik/script/prolong/mcToken.js --dry
```
Comment (caillou) [inline packages/republik/script/prolong/mcToken.js:6]:
```suggestion
 * node packages/republik/script/prolong/mcToken.js --dry
```

File: packages/republik/script/prolong/printEmails.js [hunk 19]
```diff
@@ -5,8 +5,7 @@
  * - uses prolongBeforeDate which respects cancellations, sleeping memberships and pending pledges
  *
  * Usage:
- * cd servers/republik
- * script/prolong/printEmails.js 2019-01-16
+ * /packages/republik/script/prolong/printEmails.js 2019-01-16
```
Comment (caillou) [inline packages/republik/script/prolong/printEmails.js:8]:
```suggestion
 * node /packages/republik/script/prolong/printEmails.js 2019-01-16
```

File: packages/voting/scripts/mainstreamscore/seed.js [hunk 20]
```diff
@@ -9,7 +9,7 @@ const submitAnswer = require('@orbiting/backend-modules-voting/graphql/resolvers
 const moment = require('moment')
 const { v4: uuid } = require('uuid')
 const { t } = require('@orbiting/backend-modules-translate')
-const { loaderBuilders } = require('../../../../servers/republik/server')
+const { loaderBuilders } = require('../../../../servers/graphql/server')
```
Comment (caillou) [inline packages/voting/scripts/mainstreamscore/seed.js:12]:
```suggestion
const { loaderBuilders } = require('@orbiting/graphql-server/server')
```

File: packages/test/instance/index.js [hunk 21]
```diff
@@ -79,7 +69,7 @@ const init = async ({ serverName, publicationScheduler, searchNotifyListener = n
   })
 
   // require server's server.js and start
-  const Server = require(`${relativeServerPath}server`)
+  const Server = require(`../../../servers/graphql/server`)
```
Comment (caillou) [inline packages/test/instance/index.js:72]:
```suggestion
  const Server = require(`@orbiting/graphql-server/server`)
```

**Comment by caillou** [comment]
I think you missed https://github.com/orbiting/backends/blob/master/packages/test/script/testShutdown.js#L4-L6

```js
const serverName = 'republik'

const relativeServerPath = `../../../servers/${serverName}/`
```

should be

```js
const relativeServerPath = `../../../servers/graphql/`
```

This raises the question: Why is this file there? It does not seem to be used.

File: packages/publikator/package.json (hunk 12 above)
Comment (patte) [reply packages/publikator/package.json:7]: No. But as I understand it, `private` is there to ensure that a package is never published to a registry and we don't need that from a securit standpoint.

File: packages/publikator/script/addDurations.js (hunk 13 above)
Comment (patte) [reply packages/publikator/script/addDurations.js:7]: Why should this be needed?

**Review by caillou — APPROVED** [review]

**Pull request merged**